src/components/MovieListPage/MovieListPage.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useState } from 'react'
+import { GenreSelect } from '../GenreSelect/GenreSelect.tsx'
+import SortControl from '../SortControl/SortControl.tsx'
+import { MovieTile } from '../MovieTile/MovieTile.tsx'
+import { Search } from '../Search/Search.tsx'
+import { MovieGenres } from '../../common/constants/constants.ts'
+import './MovieListPage.css'
+import { MovieDetails } from '../MovieDetails/MovieDetails.tsx'
+import { getMovies } from '../../api/Movie.ts'
+import { Movie } from '../../common/interfaces/Movie'
+import { capitalizeFirstLetter } from '../../utils/utils.ts'
+
+const MovieListPage: React.FC = () => {
+    const [searchQuery, setSearchQuery] = useState('')
+    const [sortCriterion, setSortCriterion] = useState('')
+    const [activeGenre, setActiveGenre] = useState('ALL')
+    const [selectedMovie, setSelectedMovie] = useState<Movie | null>()
+    const [movieList, setMovieList] = useState<Movie[]>([])
+    const genres = ['ALL', ...MovieGenres].map(genre => genre.toUpperCase())
```

Comment:
Is there some particular reason why you are converting the genres to upper case and then back when you need to communicate with backend? If it's only because they look uppercased on UI, I suggest that you change it with CSS instead (`text-transform: uppercase` will do the trick). Converting them back and forth adds confusion and uncertainty about the values.